src-data-processing/target_data_maker.R

```diff
@@ -0,0 +1,39 @@
+#' Target_data_maker
+#'
+#' A script to create data files with data available on a given date.
+
+require(tidyverse)
+require(arrow)
+# reading in the data; Here we are using the 100k sample to test instead of the full metadata.
+# If you want to use the full data
+# change the file path and use the commended #full_data_file <- read_delim_arrow(file_path, delim = "\t") line,
+# because of the different file types.
+file_path <- "./input-data/100Ksample.tsv.xz"
+full_data_file <- read_tsv(file_path)
+#full_data_file <- read_delim_arrow(file_path, delim = "\t")
+
+# removing any data before 2022, non-humans, and not in the 52 regions we care about
+full_data_file <- filter(full_data_file, date_submitted >= as.Date("2022-01-01"),
+                         host == "Homo sapiens", division %in% c(state.name, "Washington DC","Puerto Rico"))
+# removing all the columns we don't need
+full_data_file <- full_data_file[c("division", "Nextstrain_clade", "date", "date_submitted")]
+# removing any NA's
+full_data_file <- filter(full_data_file, !is.na(date))
+# reducing each clade name to there 3 character designation except recombinant
+full_data_file$Nextstrain_clade <- substr(full_data_file$Nextstrain_clade,1,3)
```

Comment:
to make this code more compact, readable and in standard tidyverse style you could do something like
mutate(clade = ifelse(Nextstrain_clade == "recombinant", "recombinant", substr(Nextstrain_clade, 1, 3)))